server/src/main/java/com/flipkart/varadhi/events/EntityEventDispatcher.java

```diff
@@ -0,0 +1,222 @@
+        log.debug("Processing {} event for {} {}", rawEvent.operation(), resourceType, resourceName);
+
+        try {
+            EntityEventListener<T> listener = (EntityEventListener<T>)listeners.get(resourceType);
+
+            Object rawResource = rawEvent.resource();
+            T typedResource = null;
+
+            if (rawResource != null) {
+                if (rawResource instanceof Map) {
+                    String json = JsonMapper.jsonSerialize(rawResource);
+                    Class<T> entityClass = getEntityClassForResourceType(resourceType);
+                    typedResource = JsonMapper.jsonDeserialize(json, entityClass);
+                } else if (rawResource instanceof MetaStoreEntity) {
+                    typedResource = (T)rawResource;
+                }
+            }
+
+            EntityEvent<T> typedEvent = new EntityEvent<>(
+                resourceType,
+                resourceName,
+                rawEvent.operation(),
+                typedResource,
+                rawEvent.commiter()
+            );
+
+            listener.onChange(typedEvent);
+            return CompletableFuture.completedFuture(ResponseMessage.fromPayload("OK", messageId));
+        } catch (ClassCastException e) {
+            String errorMsg = String.format("Resource type mismatch for %s of type %s", resourceName, resourceType);
+            log.error(
+                "Type mismatch for {} event on resource {}: {}",
+                rawEvent.operation(),
+                resourceType,
+                resourceName,
+                e
+            );
+            return createErrorResponse(messageId, errorMsg, e);
+        } catch (Exception e) {
+            log.error("Failed to process {} operation for {}: {}", rawEvent.operation(), resourceType, resourceName, e);
+            return createErrorResponse(messageId, e.getMessage(), e);
+        }
+    }
+
+    /**
+     * Creates an error response with the given error details.
+     * <p>
+     * This method handles various error cases and ensures proper logging.
+     *
+     * @param messageId the ID of the original message, or null if not available
+     * @param errorMsg  the error message
+     * @param cause     the cause of the error, or null if not available
+     * @return a future that completes with the error response
+     */
+    private CompletableFuture<ResponseMessage> createErrorResponse(String messageId, String errorMsg, Throwable cause) {
+        Exception exception = switch (cause) {
+            case Exception ex -> ex;
+            case null -> new IllegalArgumentException(errorMsg);
+            default -> new IllegalArgumentException(errorMsg, cause);
+        };
+
+        if (cause == null) {
+            log.error(errorMsg);
+        } else if (errorMsg.equals(cause.getMessage())) {
+            log.error(errorMsg, cause);
+        } else {
+            log.error("{}: {}", errorMsg, cause.getMessage(), cause);
+        }
```

Comment:
no logging required. I think we are logging too much at many places. This will lead to log pollution only as more and more code is added.